voter-dapp/README.md

```diff
@@ -1,6 +1,71 @@
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+# Voter dApp
 
-## Available Scripts
+This is a browser client that UMA token holders can use to vote on price requests. The dApp interacts with the deployed [DVM](https://docs.umaproject.org/uma/oracle/technical_architecture.html) code using MetaMask's browser extension. The source code was bootstrapped from [Create React App](https://github.com/facebook/create-react-app).
+
+Features currently include:
+
+- Deploy a [designated voting proxy contract](https://docs.umaproject.org/uma/oracle/voting_with_UMA_2-key_contract.html)
+- Commit an encrypted vote
+- Backup salts associated with latest encrypted votes
+- Reveal an encrypted vote
+- Retrieve rewards for resolved votes
+- View results of resolved votes
+
+Note: this project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
+## Making changes to the Voter dApp
+
+The source code for this React app can be found in `./src`. At a minimum, you must ensure that a user can go through a simple commit and reveal flow, both using a voting proxy and not using one.
+
+### Prerequisites
+
+- Run `yarn` from the root of the repo to install the monorepo's dependencies.
+- Compile contracts by running `yarn truffle compile` from the root of the repo.
+
+### Steps to generate price requests
+
+0. Start at the root of the repository (`protocol/`).
+1. Start a local blockchain on port `9545` with enough starting ETH balances and a high enough gas limit:
+   ```bash
+   ganache-cli -p 9545 -e 10000000000 -l 9000000
+   ```
+   This seeds every account with `10000000000` ETH and sets the block gas limit to 9 million wei. This tool should also
+   display all of the private keys associated with the ganache default accounts. You should keep the private keys handy
+   because you will want to import some of them into MetaMask to test the dApp locally.
+1. Open another window and deploy all contracts:
+   ```bash
+   cd core && yarn truffle migrate --reset --network test
```

Comment:
I updated all the commands throughout to use `yarn truffle` instead of `$(npm bin)/truffle` since yarn seems to resolve more reliably. I think yarn does a more dynamic resolution to avoid errors where dependencies/binaries are hoisted.